File: python/1813. Sentence Similarity III.py
class Solution:
    def areSentencesSimilar(self, sentence1: str, sentence2: str) -> bool:
        words1 = sentence1.split(" ")
        words2 = sentence2.split(" ")

        n, m = len(words1), len(words2)
        i = j = 0

        while i < len(words1) and i < len(words2) and words1[i] == words2[i]:
            i += 1
        while j < len(words1) - i and j < len(words2) - i and words1[n - 1 - j] == words2[m - 1 - j]:
            j += 1

        return i + j >= n or i + j >= m

File: rust/1813. Sentence Similarity III.rs
impl Solution {
    pub fn are_sentences_similar(sentence1: String, sentence2: String) -> bool {
        let words1: Vec<&str> = sentence1.split_whitespace().collect();
        let words2: Vec<&str> = sentence2.split_whitespace().collect();

        let (n, m) = (words1.len(), words2.len());

        let (mut i, mut j) = (0, 0);

        while i < words1.len() && i < words2.len() && words1[i] == words2[i] {
            i += 1;
        }
        while j < words1.len() && j < words2.len() && words1[n - j - 1] == words2[m - j - 1] {
            j += 1;
        }

        i + j >= n || i + j >= m
    }
}
